### PR TITLE
Use eslint-config-metarhia and fix linting issues

### PR DESCRIPTION
**Pull request by belochub**
Refs: https://github.com/metarhia/Metarhia/issues/22

### CODE REVIEW

File: lib/collector.functor.js [hunk 1]
```diff
@@ -26,12 +25,12 @@ const collect = (
...
     if (isDone) return collector;
     if (!isDistinct || !(key in data)) {
-      if (!isCount && !keys.has(key)) return;
+      if (!isCount && !keys.has(key)) return null;
       count++;
     }
     if (err) {
       collector.finalize(err, data);
-      return;
+      return null;
```
Comment (tshemsedinov) [inline lib/collector.functor.js]: `return collector;`

File: lib/chain.js [hunk 2]
```diff
@@ -11,6 +11,7 @@ const async = op => {
     case 'series': return series;
     case 'find': return find;
   }
+  return null;
```
Comment (tshemsedinov) [inline lib/chain.js]:
```js
return (items, fn, callback) => {
  return callback(null, items.map(id));
};
```
where
`const Id = x => x;`

File: lib/chain.js [hunk 3]
```diff
@@ -23,7 +24,7 @@ ArrayChain.prototype.execute = function(err) {
 
   if (!item.op) {
     if (err) throw err;
-    else return;
+    else return null;
```
Comment (tshemsedinov) [inline lib/chain.js]: :+1:

File: lib/chain.js [hunk 4]
```diff
@@ -37,7 +38,7 @@ ArrayChain.prototype.execute = function(err) {
 
   if (err) {
     this.execute(err);
-    return;
+    return null;
```
Comment (tshemsedinov) [inline lib/chain.js]: :+1:

File: lib/chain.js [hunk 5]
```diff
@@ -47,6 +48,8 @@ ArrayChain.prototype.execute = function(err) {
     const op = async(item.op);
     op(this.array, item.fn, next);
   }
+
+  return null;
```
Comment (tshemsedinov) [inline lib/chain.js]: :+1:

File: lib/composition.js [hunk 6]
```diff
@@ -152,7 +151,7 @@ Composition.prototype.sequential = function() {
 Composition.prototype.then = function(fulfill, reject) {
   if (this.canceled) {
     reject(new Error(COMPOSE_CANCELED));
-    return;
+    return this;
```
[the remark this answers is not in the document]
Comment (tshemsedinov) [reply lib/composition.js]: `this` is of for now, but we will use composition as thenable and with async/await so in future we will implement returning new composition from `Composition.prototype.then`

[the remark this answers is not in the document]
Comment (tshemsedinov) [reply lib/composition.js]: Need tests/examples and more research

File: lib/poolify.js [hunk 7]
```diff
@@ -10,15 +10,15 @@ const provide = callback => item => {
...
+  const pool = par => {
     if (Array.isArray(par)) {
       while (par.length) {
         const item = par.shift();
         const delayed = pool.delayed.shift();
         if (delayed) delayed(item);
         else pool.items.push(item);
       }
-      return;
+      return undefined;
```
Comment (tshemsedinov) [inline lib/poolify.js]: `return pool;`

File: lib/poolify.js [hunk 8]
```diff
@@ -31,10 +31,11 @@ const poolify = (factory, min, norm, max) => {
     const callback = provide(par);
     if (res) callback(res);
     else pool.delayed.push(callback);
+    return undefined;
```
Comment (tshemsedinov) [inline lib/poolify.js]: `return pool;`

File: lib/poolify.opt.js [hunk 9]
```diff
@@ -12,15 +12,15 @@ const poolify = (factory, min, norm, max) => {
...
+  return par => {
     if (Array.isArray(par)) {
       while (par.length) {
         const item = par.shift();
         const request = delayed.shift();
         if (request) request(item);
         else items.push(item);
       }
-      return;
+      return undefined;
```
Comment (tshemsedinov) [inline lib/poolify.opt.js]: `return pool;`

File: lib/poolify.opt.js [hunk 10]
```diff
@@ -33,6 +33,7 @@ const poolify = (factory, min, norm, max) => {
     const callback = provide(par);
     if (res) callback(res);
     else delayed.push(callback);
+    return undefined;
```
Comment (tshemsedinov) [inline lib/poolify.opt.js]: `return pool;`

File: lib/poolify.symbol.js [hunk 11]
```diff
@@ -16,12 +16,12 @@ const provide = callback => item => {
...
 const poolify = (factory, min, norm, max) => {
   let allocated = norm;
-  const pool = (par) => {
+  const pool = par => {
     if (par && par[poolified]) {
       const delayed = pool.delayed.shift();
       if (delayed) delayed(par);
       else pool.items.push(par);
-      return;
+      return undefined;
```
Comment (tshemsedinov) [inline lib/poolify.symbol.js]: `return pool;`

File: lib/poolify.symbol.js [hunk 12]
```diff
@@ -34,10 +34,11 @@ const poolify = (factory, min, norm, max) => {
     const callback = provide(par);
     if (res) callback(res);
     else pool.delayed.push(callback);
+    return undefined;
```
Comment (tshemsedinov) [inline lib/poolify.symbol.js]: `return pool;`

File: lib/queue.js [hunk 13]
```diff
@@ -136,6 +136,7 @@ Queue.prototype.takeNext = function(
   }
   const task = tasks.shift();
   if (task) this.next(task);
+  return this;
```
Comment (tshemsedinov) [inline lib/queue.js]: :+1:

File: lib/chain.js (hunk 2 above)
Comment (lundibundi) [reply lib/chain.js]: Why not simple
`items.map(id)` -> `items.slice()`?

Comment (tshemsedinov) [reply lib/chain.js]: Yes, `.slice(0)` will be better

File: lib/poolify.opt.js (hunk 9 above)
Comment (belochub) [reply lib/poolify.opt.js]: @tshemsedinov there is no `pool` available in this scope.

File: lib/poolify.opt.js (hunk 10 above)
Comment (belochub) [reply lib/poolify.opt.js]: @tshemsedinov there is no `pool` available in this scope.

File: lib/poolify.opt.js (hunk 9 above)
Comment (tshemsedinov) [reply lib/poolify.opt.js]: We need to put `par => {` to identifier `pool` and return it

File: lib/poolify.opt.js (hunk 10 above)
Comment (tshemsedinov) [reply lib/poolify.opt.js]: see above

File: lib/poolify.symbol.js (hunk 11 above)
Comment (tshemsedinov) [reply lib/poolify.symbol.js]: see above

**Comment by belochub** [comment]
@lundibundi @nechaido @tshemsedinov ping.

**Pull request merged**